feat(storybook): add `withPageUrl` decorator and update stories to use `pageUrl` param

rafaeelaudibert opened the pull request:
- Introduced a new decorator `withPageUrl` to set the page URL in Storybook stories.
- Updated multiple stories to utilize the new `pageUrl` parameter, enhancing navigation context.
- Removed direct router actions in stories to streamline the setup process.

This change improves the organization and usability of Storybook stories, making it easier to manage page URLs.

Yes, this was AI-generated ☝🏻 


No snapshot changes 🎉🎉🎉🎉🎉

rafaeelaudibert committed:
feat(storybook): add withPageUrl decorator and update stories to use pageUrl parameter

- Introduced a new decorator `withPageUrl` to set the page URL in Storybook stories.
- Updated multiple stories to utilize the new `pageUrl` parameter, enhancing navigation context.
- Removed direct router actions in stories to streamline the setup process.

This change improves the organization and usability of Storybook stories, making it easier to manage page URLs.

pauldambra reviewed (approved):
in principle ✅ 
snapshots failing at the mo'

also by changing so many files you've triggered robot reviewers on a bunch of files they've not seen before. there might be things its worth following up on in a different PR (since this is already pretty big)

joshsny commented on frontend/src/layout/ErrorProjectUnavailable.stories.tsx:
```diff
@@ -109,7 +108,6 @@ export const NoSelectableProjects = (): JSX.Element => {
     })
     useEffect(() => {
         loadCurrentTeamSuccess(null)
```
I am guessing there are situations where the order of operations in this `useEffect` matter, so we might not always want to use the helper

rafaeelaudibert replied:
In some cases it might fail, you're correct. People should use this by default, but they might need to use it in some situations. There are still ~10 occurences in the codebase where we're using the router directly because it doesn't fit this usecase (render App, and go to specific page)

joshsny reviewed (approved):
Looks good 😎

robbie-c commented on frontend/src/scenes/experiments/stories/ExperimentWithFunnelMetric.stories.tsx:
```diff
@@ -42,13 +40,10 @@ const meta: Meta = {
+export const ExperimentWithFunnelMetric: Story = {
+    play: async () => {
+        // Add a small delay to ensure charts render completely
+        await new Promise((resolve) => setTimeout(resolve, 500))
```
+1 to needing to fix delays, also +1 to fixing this elsewhere

Additionally, we have a `delay(500)` that we could use instead, which is a little cleaner if we're going to have a simple delay.

It's better to do something like `waitFor(() => someCheck(), 500)` but that's probably out of scope

rafaeelaudibert replied:
Yeah, I just reused what the Experiments team setup. I agree this should be improved. Flagging it here @PostHog/team-experiments, but I might take a stab at it

robbie-c commented on frontend/src/scenes/experiments/stories/ExperimentWithLegacyFunnelsQuery.stories.tsx:
```diff
@@ -42,13 +42,10 @@ const meta: Meta = {
+export const ExperimentWithLegacyFunnelsQuery: Story = {
+    play: async () => {
+        // Add a small delay to ensure charts render completely
+        await new Promise((resolve) => setTimeout(resolve, 500))
```
```suggestion
        await delay(500)
```

robbie-c replied:
(will need an import)

robbie-c reviewed (approved):
LGTM, I agree that it's worth looking at some of these AI comments in a future PR

rafaeelaudibert committed:
refactor: Use FFs from `FEATURE_FLAGS` in stories

rafaeelaudibert committed:
feat(utils): add makeDelay function and refactor stories to use it for delays

- Introduced `makeDelay` function to simplify delay handling in stories.
- Updated multiple experiment stories to utilize `makeDelay` instead of inline promises for better readability and maintainability.
- Cleaned up unnecessary async functions in stories, enhancing clarity.

rafaeelaudibert committed:
fix: Use proper url for sidepanel

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)
